### PR TITLE
Add footer and privacy note to the start dm dialog

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/17365
Fixes https://github.com/vector-im/element-web/issues/17359

image

### CODE REVIEW

**Comment by niquewoodhouse** [comment]
Great, thanks .  As we've added something to the bottom of the dialog, but it's stayed the same height, the vertical space is a bit more precious, these tweaks helped me:

- [ ] mx_InviteDialog_helpText have margin-top 0
- [ ] I gave too little margin to the mx_InviteDialog_footer h3 (12 instead of 8)
- [ ] .mx_InviteDialog_userSections margin-top 4
- [ ] .mx_InviteDialog_userSections padding-bottom 4
- [ ] Introduce a userSections class specific for the `Some results may be hidden for privacy` section with padding-top 8 and padding-bottom 16, margin-left 18 (so it lines up with 'show more')
- [ ] .mx_InviteDialog_userSections height (calc(100% - 175px)
- [ ] mx_light-fg-color border for the link field instead of input-border-color (in practice, it standing out a bit is good)

These changes seem to give us back almost all of the people/results:

image

I didn't realise the `Some results may be hidden for privacy` section would appear if you're not searching.  `Results` doesn't make sense if it's not a search, but we can just change the copy to say `suggestions` and it works fine. 

Thanks

____

@nadonomy there's some specific paddings on the invite dialog to pad the content on the left and the right, so the list of people is less wide

image

Do you mind if it changes to have no bespoke padding, would just line up a bit neater as there's a full width thing at the bottom of the dialog now.

image

File: res/css/views/dialogs/_InviteDialog.scss [hunk 1]
```diff
@@ -90,6 +98,63 @@ limitations under the License.
...
+    padding: 8px 0 16px 0;
+    font-size: $font-14px;
+
+    > span {
+        color: $primary-fg-color;
+        font-weight: 600;
+    }
+
+    > p {
+        margin:0;
```
Comment (t3chguy) [inline res/css/views/dialogs/_InviteDialog.scss:111]:
```suggestion
        margin: 0;
```

File: res/css/views/dialogs/_InviteDialog.scss [hunk 2]
```diff
@@ -212,22 +277,32 @@ limitations under the License.
...
+    }
 }
 
 .mx_InviteDialog_userSections {
-    margin-top: 10px;
+    margin-top: 4px;
     overflow-y: auto;
-    padding-right: 45px;
-    height: 455px; // mx_InviteDialog's height minus some for the upper elements
+    padding: 0px 45px 4px 0px;
```
Comment (t3chguy) [inline res/css/views/dialogs/_InviteDialog.scss:293]:
```suggestion
    padding: 0 45px 4px 0;
```

File: res/css/views/dialogs/_InviteDialog.scss [hunk 3]
```diff
@@ -212,22 +277,32 @@ limitations under the License.
...
+    height: calc(100% - 175px); // mx_InviteDialog's height minus some for the upper  and lower elements
 }
 
+
 // Right margin for the design. We could apply this to the whole dialog, but then the scrollbar
 // for the user section gets weird.
-.mx_InviteDialog_helpText,
 .mx_InviteDialog_addressBar {
-    margin-right: 45px;
+    margin: 8px 45px 0px 0px;
```
Comment (t3chguy) [inline res/css/views/dialogs/_InviteDialog.scss:301]:
```suggestion
    margin: 8px 45px 0 0;
```

File: res/css/views/dialogs/_InviteDialog.scss [hunk 4]
```diff
@@ -212,22 +277,32 @@ limitations under the License.
...
+}
+
+.mx_InviteDialog_helpText {
+    margin:0px;
```
Comment (t3chguy) [inline res/css/views/dialogs/_InviteDialog.scss:305]:
```suggestion
    margin: 0;
```

**Comment by niquewoodhouse** [comment]
**Review**

- The 'copied' tooltip/feedback seems to stay visible after you close the dialog
- This change appears to impact the size of the non-DM invite dialog (eg invite to this room)
image

Can that be undone easily enough as it wasn't intended as part of this papercut?

**Comment by turt2live** [comment]
@t3chguy can we get screenshots of what it looks like in code to the PR description please?

**Review by niquewoodhouse — APPROVED** [review]
lgtm, thanks

File: src/components/views/dialogs/InviteDialog.tsx [hunk 5]
```diff
@@ -1238,6 +1249,25 @@ export default class InviteDialog extends React.PureComponent<IInviteDialogProps
         }
     }
 
+    async onLinkClick(e) {
```
Comment (turt2live) [inline src/components/views/dialogs/InviteDialog.tsx:1252]:
```suggestion
    public async onLinkClick(e) {
```

or whatever member access was intended.

File: src/components/views/dialogs/InviteDialog.tsx [hunk 6]
```diff
@@ -1238,6 +1249,25 @@ export default class InviteDialog extends React.PureComponent<IInviteDialogProps
...
+        e.preventDefault();
+        selectText(e.target);
+    }
+
+    onCopyClick = async e => {
```
Comment (turt2live) [inline src/components/views/dialogs/InviteDialog.tsx:1257]:
```suggestion
    public onCopyClick = async e => {
```

or whatever member access was intended.

File: src/components/views/dialogs/InviteDialog.tsx [hunk 7]
```diff
@@ -1316,6 +1346,26 @@ export default class InviteDialog extends React.PureComponent<IInviteDialogProps
             }
             buttonText = _t("Go");
             goButtonFn = this.startDm;
+            extraSection = <div className="mx_InviteDialog_section_hidden_suggestions_disclaimer">
+                <span>{ _t("Some suggestions may be hidden for privacy.") }</span>
+                <p>{ _t("If you can’t see who you’re looking for, send them your invite link below.") }</p>
```
Comment (turt2live) [inline src/components/views/dialogs/InviteDialog.tsx:1351]:
```suggestion
                <p>{ _t("If you can't see who you’re looking for, send them your invite link below.") }</p>
```

**Review by turt2live — CHANGES_REQUESTED** [review]
screenshots in the PR description please to make CSS review easier :)

File: src/components/views/dialogs/InviteDialog.tsx (hunk 7 above)
Comment (t3chguy) [reply src/components/views/dialogs/InviteDialog.tsx:1351]: Serves me right for using the Copy button in Figma

**Review by turt2live — APPROVED** [review]

**Pull request merged**